modules/gapi/src/backends/common/gbackend.hpp

```diff
@@ -54,6 +54,20 @@ namespace gimpl {
         virtual cv::GMatDesc desc() const override { return cv::descr_of(m_mat); }
     };
 
+    class RMatMediaFrameAdapter : public RMat::Adapter {
```

Comment:
Check `RMatMediaAdapterBGR` in `gbackend.hpp`

Reply:
I have checked it and relied on it, but this `RMatMediaAdapterBGR` holds `View`, not `MediaFrame` inside and I am not really sure that the problem is in that that `View` gets destructed. `View` doesn't hold origin data, it just holds pointers to it, so I think I need to hold `MediaFrame` with adapter inside to preserve data until `RMat` will be destructed.

Reply:
Reworked as discussed